NLog.Web.AspNetCore/LayoutRenderers/AspNetRequestFormLayoutRenderer.cs

```diff
@@ -0,0 +1,91 @@
+﻿using NLog.LayoutRenderers;
+using NLog.Web.Internal;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace NLog.Web.LayoutRenderers
+{
+    /// <summary>
+    /// ASP.NET Request Form Data
+    /// </summary>
+    /// <example>
+    /// <para>Example usage of ${aspnet-request-form}:</para>
+    /// <code lang="NLog Layout Renderer">
+    /// ${aspnet-request-form} - Produces - All Form Data from the Request with each key/value pair separated by a new line.
+    /// ${aspnet-request-form:Include=id,name} - Produces - Only Form Data from the Request with keys "id" and "name".
+    /// ${aspnet-request-form:Exclude=id,name} - Produces - All Form Data from the Request except the keys "id" and "name".
+    /// ${aspnet-request-form:Include=id,name:Exclude=id} - Produces - Only Form Data from the Request with key "name" (<see cref="Exclude"/> takes precedence over <see cref="Include"/>).
+    /// ${aspnet-request-form:Separator= / } - Produces - All Form Data from the Request with each key/value pair separated by " / ".
+    /// </code>
+    /// </example>
+    [LayoutRenderer("aspnet-request-form")]
+    public class AspNetRequestFormLayoutRenderer : AspNetLayoutRendererBase
+    {
+        /// <summary>
+        /// Gets or sets the form keys to include in the output.  If omitted, all are included.  <see cref="Exclude"/> takes precedence over <see cref="Include"/>.
+        /// </summary>
+        /// <docgen category='Rendering Options' order='10' />
+#if ASP_NET_CORE
+        public ISet<string> Include { get; set; }
+#else
+        public HashSet<string> Include { get; set; }
+#endif
+
+        /// <summary>
+        /// Gets or sets the form keys to exclude from the output.  If omitted, none are excluded.  <see cref="Exclude"/> takes precedence over <see cref="Include"/>.
+        /// </summary>
+        /// <docgen category='Rendering Options' order='10' />
+#if ASP_NET_CORE
+        public ISet<string> Exclude { get; set; }
+#else
+        public HashSet<string> Exclude { get; set; }
+#endif
+
+        /// <summary>
+        /// Gets or sets the separator to use between each key/value pair.  If omitted, <see cref="Environment.NewLine"/> is used.
+        /// </summary>
+        /// <docgen category='Rendering Options' order='10' />
+        public string Separator { get; set; }
```

Comment:
Maybe this should be a Layout. What do you think? We can use then `${newline}` etc

Reply:
That would be better.  I didn't realize I could do that.  Do you think I should still have newline as the default separator or would something else be better as the default?  The main reason I had newline as the default was because I didn't know I could set it up to take a layout.  Thanks!

Reply:
I think newline is fine, but in other renderers we use space. Not sure what would be the best in this case.

Reply:
I will change it to a space to keep it consistent with the others.   With it being a Layout instead of a string people can easily make it a newline if they want.

Reply:
I just noticed the base class `AspNetLayoutMultiValueRendererBase` and I'm wondering if I should change this to extend that class.

Reply:
doubting about that. It isn't a bad idea, but unfortunately no Layouts there.  Both are fine to me.

Reply:
OK.  I will do that now.

Reply:
Thanks!